pkgs/sketch_pad/lib/main.dart

```diff
@@ -217,16 +218,28 @@ class DartPadMainPage extends StatefulWidget {
   State<DartPadMainPage> createState() => _DartPadMainPageState();
 }
 
-class _DartPadMainPageState extends State<DartPadMainPage> {
+class _DartPadMainPageState extends State<DartPadMainPage>
+    with SingleTickerProviderStateMixin {
   late final SplitViewController mainSplitter;
 
   late AppModel appModel;
   late AppServices appServices;
+  late final TabController tabController;
+  final ValueKey<String> _executionWidgetKey =
+      const ValueKey('execution-widget');
+  final ValueKey<String> _loadingOverlayKey =
+      const ValueKey('loading-overlay-widget');
+  final ValueKey<String> _editorKey = const ValueKey('editor');
+  final ValueKey<String> _consoleKey = const ValueKey('console');
 
   @override
   void initState() {
     super.initState();
 
+    tabController = TabController(length: 3, vsync: this)
+      ..addListener(() {
+        setState(() {});
```

Comment:
Can you add a comment here? This causes a rebuild on tab selected index changes?